jib-cli/src/test/java/com/google/cloud/tools/jib/cli/cli2/JibCliTest.java

```diff
@@ -329,83 +329,66 @@ public void testParse_toCredentialHelperAndFromUsernamePassword() {
-          });
-    }
-
-    @Parameterized.Parameter(0)
-    public String usernameField;
-
-    @Parameterized.Parameter(1)
-    public String passwordField;
-
-    @Test
-    public void testParse_usernameWithoutPassword() {
-      MissingParameterException mpe =
-          assertThrows(
-              MissingParameterException.class,
-              () ->
-                  CommandLine.populateCommand(
-                      new JibCli(), "--target", "test-image-ref", usernameField, "test-username"));
-      assertThat(mpe.getMessage())
-          .isEqualTo("Error: Missing required argument(s): " + passwordField);
-    }
-
-    @Test
-    public void testParse_passwordWithoutUsername() {
-      MissingParameterException mpe =
-          assertThrows(
-              MissingParameterException.class,
-              () ->
-                  CommandLine.populateCommand(
-                      new JibCli(), "--target", "test-image-ref", passwordField, "test-password"));
-      assertThat(mpe.getMessage())
-          .isEqualTo("Error: Missing required argument(s): " + usernameField + "=<username>");
-    }
+  private Object usernamePasswordPairs() {
+    return new Object[][] {
+      {"--username", "--password"},
+      {"--to-username", "--to-password"},
+      {"--from-username", "--from-password"}
+    };
   }
 
-  @RunWith(Parameterized.class)
-  public static class IncompatibleCredentialCombos {
-    @Parameterized.Parameters
-    public static Collection<String[][]> data() {
-      return Arrays.asList(
-          new String[][][] {
-            {{"--credential-helper=x", "--to-credential-helper=x"}},
-            {{"--credential-helper=x", "--from-credential-helper=x"}},
-            {{"--credential-helper=x", "--username=x", "--password=x"}},
-            {{"--credential-helper=x", "--from-username=x", "--from-password=x"}},
-            {{"--credential-helper=x", "--to-username=x", "--to-password=x"}},
-            {{"--username=x", "--password=x", "--from-username=x", "--from-password=x"}},
-            {{"--username=x", "--password=x", "--to-username=x", "--to-password=x"}},
-            {{"--username=x", "--password=x", "--to-credential-helper=x"}},
-            {{"--username=x", "--password=x", "--from-credential-helper=x"}},
-            {{"--from-credential-helper=x", "--from-username=x", "--from-password=x"}},
-            {{"--to-credential-helper=x", "--to-password=x", "--to-username=x"}},
-          });
-    }
-
-    @Parameterized.Parameter public String[] authArgs;
-    private final String[] requiredArgs = new String[] {"--target=ignored"};
-
-    @Test
-    public void testParse_usernameWithoutPassword() {
-      MutuallyExclusiveArgsException meae =
-          assertThrows(
-              MutuallyExclusiveArgsException.class,
-              () ->
-                  CommandLine.populateCommand(
-                      new JibCli(), ArrayUtils.addAll(requiredArgs, authArgs)));
```

Comment:
QQ: which dependency has this util?

Reply:
commons-text (jib-cli) pulls in commons-lang